patches/minecraft/net/minecraft/block/Block.java.patch

```diff
@@ -1504,12 +1504,30 @@
 +        return isBurning(world, pos) ? net.minecraft.pathfinding.PathNodeType.DAMAGE_FIRE : null;
 +    }
 +
++    /**
++     * @param blockState The state for this block
++     * @param world The world this block is in
++     * @param pos The position of this block
++     * @param side The side of this block that the chest lid is trying to open into
++     * @return true if the chest should be prevented from opening by this block
++     */
++    public boolean doesSideBlockChestOpening(IBlockState blockState, IBlockAccess world, BlockPos pos, EnumFacing side)
```

Comment:
Do we need a side parameter here?

Reply:
Maybe you have a chest that opens sideways? I dunno, figured I'd keep the option open.

Reply:
Please keep,
I have a chest that opens sideways, and another that can be rotated to open in any direction.

Reply:
I also advise keeping it, but for the sake of vanilla shulker boxes.

Reply:
The shulker box logic is totally different, strangely. It uses a bounding box collision test.

Reply:
Drawers from mods may want to open sideways, too.

Reply:
This really only applies for custom chests, though. Other things should probably follow the shulker box example.

As someone does actually have a chest that opens sideways, it's fine to keep this.